File: app/src/main/java/me/liuyun/bjutlgn/ui/StatusLockedActivity.java
package me.liuyun.bjutlgn.ui;

import android.app.Dialog;
import android.content.DialogInterface;
import android.net.CaptivePortal;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.WindowManager;


public class StatusLockedActivity extends AppCompatActivity implements DialogInterface.OnDismissListener {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        CaptivePortal captivePortal = getIntent().getParcelableExtra(ConnectivityManager.EXTRA_CAPTIVE_PORTAL);

        getWindow().addFlags(WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        Dialog dialog = StatusDialog.statusDialog(this, captivePortal);
        dialog.setOnDismissListener(this);
        dialog.show();
    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        finish();
    }
}

File: app/src/main/kotlin/me/liuyun/bjutlgn/ui/StatusLockedActivity.kt
package me.liuyun.bjutlgn.ui

import android.content.DialogInterface
import android.net.CaptivePortal
import android.net.ConnectivityManager
import android.os.Bundle
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity

class StatusLockedActivity : AppCompatActivity(), DialogInterface.OnDismissListener {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val captivePortal = intent.getParcelableExtra<CaptivePortal>(ConnectivityManager.EXTRA_CAPTIVE_PORTAL)

        setShowWhenLocked(true)
        setTurnScreenOn(true)
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
        window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
        val dialog = StatusDialog.statusDialog(this, captivePortal)
        dialog.setOnDismissListener(this)
        dialog.show()
    }

    override fun onDismiss(dialog: DialogInterface) {
        finish()
    }
}
